pymatgen/analysis/spectra/spectrum.py

```diff
@@ -0,0 +1,161 @@
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License.
+
+from monty.json import MSONable
+import abc
+from scipy.interpolate import interp1d
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+from pymatgen import Structure
+import json
+import numpy as np
+
+"""
+This module defines classes to represent all spectra
+"""
+
+__author__ = "Chen Zheng"
+__copyright__ = "Copyright 2012, The Materials Project"
+__version__ = "2.0"
+__maintainer__ = "Chen Zheng"
+__email__ = "chz022@ucsd.edu"
+__date__ = "Aug 9, 2017"
+
+
+class Spectrum(abc.ABC):
+    """
+    Base class for Xanespectrum and Exafspectrum.
+    Not meant to be instantiated directly
+    """
+
+    @property
+    def spectrum_structure(self):
+        """
+        Returns the structure associated with the spectrum
+        """
+        return self.structure
+
+    @property
+    def spectrum_type(self):
+        """
+        Returns the spectrum type
+        """
+        return self.type
+
+    @property
+    def spectrum_data(self):
+        """
+        Returns the spectrum
+        """
+        return self.spectrum
+
+    @classmethod
+    @abc.abstractmethod
+    def from_file(cls, filename):
+        """
+        Initiate Spectrum from a filename
+        """
+        pass
+
+
+class Xanespectrum(Spectrum, MSONable):
+    def __init__(self, formula, absorption_specie, edge, structure, spectrum):
+        """
+        Create a XANES spectrum object
+        :param formula: The reduced formula of the structure associated with the spectrum
+        :param absorption_specie: absorption specie of the spectrum
+        :param edge: absorption spectrum edge
+        :param structure: materials structure associated with the spectrum, need to be a pymatgen structure object
+        :param spectrum: Spectrum data
+        """
+
+        self.formula = formula
+        self.absorption_specie = absorption_specie
+        self.edge = edge
+        self.structure = structure
+        self.spectrum = spectrum
+
+
+class XanesFEFF(Xanespectrum):
```

Comment:
Indeed, the XanesFEFF object has some specific information related to the db entry we generated in our database construction. The absorbing_atom_index field and mp-id field value we extracted from the XAS database entry's are pretty useful in generating ensemble spectrum. Some functions we tend to implement regarding spectrum processing rely on these few field. Intrinsically, those specific fields' information more db related. Also, some input parameters like S02 might be helpful in later analysis, I will keep the XANES class as a generic. For the more db specific spectrum classes construction, should I inherited the XANES class from pymatgen and keep them in private repo? Thanks a lot,